**Convert the fasta file of variant sites within the inversion to a SNAPP XML file**

There are two ways to create an XML file from a fasta. The first is to use the BEAST2 software [BEAUti](https://github.com/CompEvol/beast2/tree/master/src/beast/app/beauti).

The second is to use the file [500fasta2SNAPPxml.pl](https://github.com/katlande/Helianthus_Inversions_Dating/blob/master/fasta_to_xml/500fasta2SNAPPxml.pl) generated by [Greg Owens](https://github.com/owensgl/wild_gwas_2018/blob/master/perl_tools/fasta2SNAPPxml.pl) to convert the [example fasta](https://github.com/katlande/Helianthus_Inversions_Dating/blob/master/vcf_to_fasta/example.fa) 
into the [example xml](https://github.com/katlande/Helianthus_Inversions_Dating/blob/master/fasta_to_xml/example.xml) with the code `perl 500fasta2SNAPPxml.pl example.fa hapoltype.txt > example.xml`.

The output file will be a ready-to-analyze XML file wherein each individual is sorted by haplotype.
- [haplotype.txt](https://github.com/katlande/Helianthus_Inversions_Dating/blob/master/fasta_to_xml/haplotype.txt) is a list of the haplotypes of each individual in the fasta

**NOTE: this code generates an XML file with priors specific to Helianthus dating. For other lineages, the perl file *500fasta2SNAPPxml.pl***
**would need to be modified, or XMLs can be generated from fastas using**
